core/libp2p/basic/closer.hpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CLOSER_HPP
+#define KAGOME_CLOSER_HPP
+
+#include <outcome/outcome.hpp>
+
+namespace libp2p::basic {
+
+class Closer {
```

Comment:
Closer sounds like a tool for closing something, but it has no method receiving object to close. It seems to be an interface for something closeable. Maybe Closeable would be better name?